From: Branden Robinson <branden@debian.org>
Subject: [PATCH] Add some more log and debug messages

Add log messages for xdm startup, shutdown, and normal exit.  Add a
debugging message when xdm enters its main loop.
---
 dm.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

Index: xdm/dm.c
===================================================================
--- xdm.orig/dm.c
+++ xdm/dm.c
@@ -160,6 +160,8 @@
 	exit (1);
     }
 
+    LogInfo ("Starting\n");
+
     if (nofork_session == 0) {
 	/* Clean up any old Authorization files */
 	/* AUD: all good? */
@@ -199,6 +201,7 @@
 #ifndef UNRELIABLE_SIGNALS
     (void) Signal (SIGCHLD, ChildNotify);
 #endif
+    Debug ("startup successful; entering main loop\n");
     while (
 #ifdef XDMCP
 	   AnyWellKnownSockets() ||
@@ -217,6 +220,7 @@
 #endif
     }
     Debug ("Nothing left to do, exiting\n");
+    LogInfo ("Exiting\n");
     exit(0);
     /*NOTREACHED*/
 }
@@ -387,6 +391,7 @@
 	return;
     }
     Debug ("Shutting down entire manager\n");
+    LogInfo ("Shutting down\n");
 #ifdef XDMCP
     DestroyWellKnownSockets ();
 #endif
@@ -711,6 +716,7 @@
     pid_t	pid;
 
     Debug ("StartDisplay %s\n", d->name);
+    LogInfo ("Starting X server on %s\n", d->name);
     LoadServerResources (d);
     if (d->displayType.location == Local)
     {
